service/src/main/java/uk/nhs/adaptors/gp2gp/common/service/ConfidentialityService.java

```diff
@@ -0,0 +1,46 @@
+package uk.nhs.adaptors.gp2gp.common.service;
+
+import lombok.RequiredArgsConstructor;
+import org.hl7.fhir.dstu3.model.Coding;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.nhs.adaptors.gp2gp.common.configuration.RedactionsContext;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class ConfidentialityService {
+    private static final String REDACTION_INTERACTION_ID = "RCMR_IN030000UK07";
+    private static final String REDACTION_CODE_SYSTEM = "http://hl7.org/fhir/v3/ActCode";
+    public static final String NOPAT = "NOPAT";
+    private static final String REDACTION_CONFIDENTIALITY_CODE = """
+        <confidentialityCode
+            code="NOPAT"
+            codeSystem="2.16.840.1.113883.4.642.3.47"
+            displayName="no disclosure to patient, family or caregivers without attending provider's authorization"
+        />""";
+
+    private final RedactionsContext redactionsContext;
+
+    public Optional<String> generateConfidentialityCode(List<Coding> metaSecurity) {
```

Comment:
I don't have strong opinions on this, but I wonder if passing in the whole resource would make the API a little nicer from the calling side.

i.e. if the method signature was

```suggestion
    public Optional<String> generateConfidentialityCode(Resource resource) {
```

Then when called, the could would be:

```java
    generateConfidentialityCode(observation);
```

vs

```java
    generateConfidentialityCode(observation.getMeta().getSecurity());
```

Reply:
I can do this